source/mesh_deformation/fastscapecc.cc

```diff
@@ -0,0 +1,853 @@
+              MPI_Probe(p, 42, this->get_mpi_communicator(), &status);
+              int incoming_size = 0;
+              MPI_Get_count(&status, MPI_DOUBLE, &incoming_size);
+
+              for (unsigned int i = 0; i < temporary_variables.size(); ++i)
+                {
+                  temporary_variables[i].resize(incoming_size);
+                }
+
+              for (unsigned int i = 0; i < temporary_variables.size(); ++i)
+                MPI_Recv(&temporary_variables[i][0], incoming_size, MPI_DOUBLE, p, 42, this->get_mpi_communicator(), &status);
+
+              for (unsigned int i = 0; i < temporary_variables[1].size(); ++i)
+                {
+                  int index = static_cast<int>(temporary_variables[1][i]);
+                  h[index] = temporary_variables[0][i];
+                  vz[index] = temporary_variables[2][i];
+                }
+            }
+
+          for (unsigned int i = 0; i < n_grid_nodes; ++i)
+            {
+              h_old[i] = h[i];
+            }
+
+          std::cout<<"here it works 11"<<std::endl;
+
+          const double aspect_timestep_in_years = this->get_timestep() / year_in_seconds;
+
+          unsigned int fastscape_iterations = fastscape_steps_per_aspect_step;
+          double fastscape_timestep_in_years = aspect_timestep_in_years / fastscape_iterations;
+          while (fastscape_timestep_in_years > maximum_fastscape_timestep)
+            {
+              fastscape_iterations *= 2;
+              fastscape_timestep_in_years *= 0.5;
+            }
+
+          std::cout << "[DEBUG] fastscape_steps_per_aspect_step = " << fastscape_steps_per_aspect_step << std::endl;
+          std::cout << "[DEBUG] aspect_timestep_in_years = " << aspect_timestep_in_years << std::endl;
+          std::cout << "[DEBUG] fastscape_iterations = " << fastscape_iterations << std::endl;
+            
+
+          double cell_area = surface_mesh.begin_active()->measure();
+
+          // 1. Create the FastScape grid adapter
+          auto grid = GridAdapterType(const_cast<SurfaceMeshType &>(surface_mesh), cell_area);
+          auto shape = grid.shape();  // Should be {n_grid_nodes}
+
+          // 2. Define the node status array with correct shape and type
+          xt::xtensor<fastscapelib::node_status, 1> node_status_array(shape);
+          std::fill(node_status_array.begin(), node_status_array.end(), fastscapelib::node_status::core);
+          grid.set_nodes_status(node_status_array);
+
+          // 3. Create the flow graph
+          auto flow_graph = FlowGraphType(
+            grid,
+            { fastscapelib::single_flow_router() }
+          );
+
+          // 4. Set base level nodes
+          std::vector<std::size_t> base_level_nodes = {0};
+          flow_graph.set_base_levels(base_level_nodes);
+
+          // 5. Build the erosion model
+          auto spl_eroder = fastscapelib::spl_eroder<FlowGraphType>(
+            fastscapelib::make_spl_eroder(flow_graph, kff, n, m, kdd)
```

Comment:
`kdd` is not bedrock diffusivity. [`make_spl_eroder`](https://fastscapelib.readthedocs.io/en/latest/api_cpp/eroder.html#_CPPv4I00EN12fastscapelib15make_spl_eroderE10spl_eroderI2FGER2FGRR1Kddd) creates an eroder class for Stream Power Law [channel erosion](https://fastscapelib.readthedocs.io/en/latest/api_cpp/eroder.html#channel-erosion). The last parameter controls the Newton-Raphson convergence for the non-linear case n != 1 (`slope_exp` parameter).

Actually there's no hillslope diffusion here.